actionview/lib/action_view/renderer/template_renderer.rb

```diff
@@ -3,6 +3,13 @@
 module ActionView
   class TemplateRenderer < AbstractRenderer #:nodoc:
     def render(context, options)
+      options.assert_valid_keys(
```

Comment:
Should we insert the validations higher in the stack? Or have a separate `renderable?` method?

Reply:
I'm not sure where higher this could go (or what that would gain). Immediately above this is [renderer/renderer.rb](https://github.com/rails/rails/blob/master/actionview/lib/action_view/renderer/renderer.rb) which feels like it would be an awkward place to do this validation, and we wouldn't want to go higher than that (where we wouldn't even know if it was a partial or template renderer we'd call). My thinking was that this is the closest to where we're actually using the options and that hopefully a backtrace cleaner would be enough to point to the actual caller in the app.